lib/auth.ts

```diff
@@ -111,6 +133,11 @@ export class Auth {
     return token;
   };
```

Comment:
Not directly scope of this ticket, but do you think we can simplify this e.g. with sth. like this?
```
return this._token = this.oauth2Client.createToken(..._.compact(accessToken, refreshToken, tokenType), {});
```

Reply:
You are right. Seems there are a couple of open issues for this problem:
https://github.com/microsoft/TypeScript/issues/22061
https://github.com/microsoft/TypeScript/issues/22112
https://github.com/microsoft/TypeScript/issues/5453

Proposed workaround might look like this, but I would not use it here, seems it brings more confusion than it is worth.
```typescript
const method = this.oauth2Client.createToken;
this.oauth2Client.createToken(
  ..._.compact([accessToken, refreshToken, tokenType, {}]) 
  as Parameters<typeof method>
);```